### DISCUSSION TITLE
Consider official Deno devcontainer feature

### DISCUSSION THREAD

**Discussion by jcbhmr** [denoland/deno#18144] [Ideas]
Hello! 👋

I'm interested in seeing if this @denoland organization would be **willing to _consider_** a denoland/devcontainer-features repository to house an official Deno [devcontainer feature](https://code.visualstudio.com/blogs/2022/09/15/dev-container-features) for [GitHub Codespaces](https://github.com/features/codespaces) (and other devcontainer environments).

Example:

```jsonc
// .devcontainer/devcontainer.json
{
  "features": {
    "ghcr.io/denoland/devcontainer-features/deno": {}
  }
}
```

In the current official [devcontainers/features](https://github.com/devcontainers/features) repository, they are **not accepting new features** and are instead [encouraging users to make their own](https://github.com/devcontainers/feature-starter). I think it would be beneficial to follow in the footsteps of other organizations like [Julia](https://github.com/JuliaLang/devcontainer-features), [R](https://github.com/rocker-org/devcontainer-features), [Mamba](https://github.com/mamba-org/devcontainer-features), and others by creating and maintaining an official Deno installer feature.

Deno already has its own [denoland/setup-deno](https://github.com/denoland/setup-deno) GitHub Action that is under the official umbrella, and I think having a similar repository for devcontainer features would be helpful for the community.

If this isn't tenable, that's OK! 👍 It's possible (and even probable) that there will be some kind of @devcontainers-community features repository that could house community-driven features. It'd just be nice if it were under the official umbrella.

Stemmed from [devcontainers-community#1](https://github.com/orgs/devcontainers-community/discussions/1#discussioncomment-5278444)

**Comment by mwmcode** [comment]
👍

[The development containers docs](https://docs.deno.com/runtime/reference/vscode/#using-a-development-container) seem outdated?
- ~~Remote-Containers~~ `Dev. Containers`: Add Development Container Configuration File  
- The only Deno options available are ones by [rsm-hcd](https://github.com/rsm-hcd/devcontainer-templates/tree/main/src/deno) (not official).

We can either follow @jcbhmr's suggestions or at least update the docs with steps on how to do it manually using the [official Deno docker images](https://github.com/denoland/deno_docker). Example: creating [Deno Vite-React app](https://docs.deno.com/deploy/tutorials/vite/) in a development container:

> .devcontainer/devcontainer.json
```json
{
  "name": "Deno",
  "image": "denoland/deno",
  "customizations": {
    "vscode": {
      "settings": {},
      "extensions": ["denoland.vscode-deno"]
    }
  },
  "postCreateCommand": "deno install"
}
```
